python/pyspark/sql/session.py

```diff
@@ -276,8 +312,10 @@ def getOrCreate(self) -> "SparkSession":
                     ).applyModifiableSettings(session._jsparkSession, self._options)
                 return session
 
-    builder = Builder()
-    """A class attribute having a :class:`Builder` to construct :class:`SparkSession` instances."""
+    @classproperty
```

Comment:
I think we should probably just refer to the official Python docs e.g., https://docs.python.org/3/howto/descriptor.html#class-methods

Meaning that you might just override `__get__` in `SparkSession`. In addition, we should probably have to add some comment that this codes can be removed when we drop Python 3.8 as a classmethod and property could be chained together from Python 3.9.

Reply:
I did not know that Python 3.9 supports classmethod + property. 
I will make a change to better explain this.
I'm not sure I will be able to make this depend on the Python version, but I'll try.

Reply:
Okay, I pushed a new commit on the branch.